File: tools/pdf-extract/extract.js
const fs = require('fs');
const path = require('path');
const pdfModule = require('pdf-parse');
console.error('pdfModule keys:', Object.keys(pdfModule));
console.error('pdfModule type:', typeof pdfModule);
const pdf = pdfModule.default || pdfModule;
if (pdfModule.PDFParse) {
  console.error('PDFParse prototype keys:', Object.getOwnPropertyNames(pdfModule.PDFParse.prototype));
}

async function extract(input, output) {
  const data = fs.readFileSync(input);
  const parser = new pdfModule.PDFParse();
  await parser.load(data);
  const text = parser.getText();
  fs.writeFileSync(output, text, 'utf8');
  console.log('Wrote', output);
}

(async () => {
  const base = path.resolve(__dirname, '..', '..');
  const files = [
    ['reference/dk-kleinschmidt.pdf', 'dk-kleinschmidt.txt'],
    ['reference/Checklist_Document__DK_Kleinschmidt__14501972.pdf', 'Checklist_Document__DK_Kleinschmidt__14501972.txt']
  ].map(([a,b]) => [path.resolve(base, a), path.resolve(__dirname, b)]);

  for (const [inp, out] of files) {
    console.log('Extracting', inp);
    await extract(inp, out);
  }
})();

File: tools/pdf-extract/extract.py
import sys
from pathlib import Path
from pdfminer.high_level import extract_text

base = Path(__file__).resolve().parents[2]
files = [
    (base / 'reference' / 'dk-kleinschmidt.pdf', Path(__file__).resolve().parent / 'dk-kleinschmidt.txt'),
    (base / 'reference' / 'Checklist_Document__DK_Kleinschmidt__14501972.pdf', Path(__file__).resolve().parent / 'Checklist_Document__DK_Kleinschmidt__14501972.txt')
]

for inp, out in files:
    print('Extracting', inp)
    try:
        text = extract_text(str(inp))
        out.write_text(text, encoding='utf-8')
        print('Wrote', out)
    except Exception as e:
        print('Error extracting', inp, e)
        sys.exit(1)
